distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso

```diff
@@ -70,6 +70,21 @@ type Table_Column_Helper
         problem_builder.attach_problems_before on_problems <|
             if error_on_empty && result.is_empty then problem_builder.raise_no_output_columns_with_cause else result
 
+    ## PRIVATE
+       A helper to returns a set of columns filtered by the type of the column.
+
+       Arguments:
+       - types: The types of columns to select.
+       - strict: If `True`, only columns with exactly the specified types will
+         be selected. If `False`, columns with types that are subtypes of the
+         specified types will also be selected (i.e. ignore size, precision).
```

Comment:
I don't like the phrasing 'are subtypes of the specified types' here. It is misleading because it is not consistent with what _actually_ happens. Looking at the code we check the `Value_Type` constructor - i.e. we accept 'sibling' types.

What actually happens differs from the 'subtyping' relation:
- The subtyping relation is not symmetric but actually anti-symmetric.
- If we say that 16-bit Integer is a subtype of 64-bit Integer (because a smaller int can be given where a larger int is expected, through widening), we cannot say conversely that 64-bit Integer is a subtype of 16-bit Integer.
	- But the 'sibling' relation defined by `is_same_type` **is symmetric**. So that is inconsistent.
- Moreover, in general we have conversions that make Integer kind-of-a subtype of Float, as it can be put (mostly) anywhere where a Float would be expected. But the `is_same_type` makes them distinct (rightly, I think!).

I think we need a different name for this than subtyping to avoid potential confusion.

Reply:
```suggestion
         be selected. If `False`, columns with related types will also be selected (i.e. ignore size, precision).
```